app/models/wallet.py

```diff
@@ -41,3 +44,59 @@ class IndyCredInfo(IndyCredInfoAcaPy):
 
 class CredInfoList(BaseModel):
     results: Optional[List[IndyCredInfo]] = None
+
+
+class DIDCreate(BaseModel):
+    method: Optional[StrictStr] = Field(
+        default="sov",
+        description="Method for the requested DID. Supported methods are 'sov', `web`, `did:peer:2` or `did:peer:4`.",
+        examples=["sov", "key", "did:peer:2", "did:peer:4"],
+    )
+    options: Optional[DIDCreateOptions] = Field(
+        default=None,
+        deprecated=deprecated("Please use key_type and did fields directly instead."),
+        description="To define a key type and/or a did depending on chosen DID method.",
+        examples=[{"key_type": "ed25519", "did": "did:peer:2"}],
+    )
+    seed: Optional[StrictStr] = Field(
+        default=None,
+        description="Optional seed to use for DID. Must be enabled in configuration before use.",
+    )
+    key_type: Optional[StrictStr] = Field(
+        default="ed25519",
+        description="Key type to use for the DID key_pair. Validated with the chosen DID method's supported key types.",
+        examples=["ed25519", "x25519", "bls12381g1", "bls12381g2", "bls12381g1g2"],
+    )
+    did: Optional[str] = Field(
+        default=None,
+        description="Specify final value of did (including did:<method>: prefix) if the method supports/requires it.",
+        strict=True,
+    )
+
+    @model_validator(mode="before")
+    @classmethod
+    def handle_deprecated_options(cls, values: dict) -> dict:
+        """
+        Handle both deprecated options field and new flattened fields.
+        Priority: If both are provided, new fields take precedence.
+        """
+        options = values.get("options")
+
+        if options:
+            warnings.warn(
+                "The 'options' field is deprecated. Please use 'key_type' and 'did' fields directly.",
+                DeprecationWarning,
+                stacklevel=2,
+            )
```

Comment:
This will just log a deprecation warning in our stack, which I guess we don't want. We want the user to know it's deprecated, so marking in the model (and our changelogs) should be enough